### PR TITLE
update with latest boards from atsamd-rs

### PR DESCRIPTION
**Pull request by TDHolmes**
Additionally, the table of contents automatically reformatted with my markdown formatter. I like the change, but I can revert if folks want.

https://github.com/atsamd-rs/atsamd/issues/511

[rendered](https://github.com/TDHolmes/awesome-embedded-rust/blob/atsamd-rs-boards/README.md)

### CODE REVIEW

**Comment by adamgreig** [comment]
Having the ToC use `-` instead of `*` makes sense for consistency with the rest of the file, and it's now sorted to match the content which is an improvement, but I don't like that it's now all nested under a "Embedded Rust" top-level heading (I appreciate it's because that's the top title in the document, but I don't think it's very helpful for the ToC). Maybe we could just remove the top heading and de-indent the rest?

**Comment by TDHolmes** [comment]
done. VS Code extension Markdown All in One btw, and change the setting `markdown.extension.toc.levels` from `1..6` to `2..6` to exclude the top level heading

**Comment by TDHolmes** [comment]
oh let me add the atsamd chat room too. one sec

**Review by adamgreig — APPROVED** [review]
Great, thanks for the update!

bors merge

File: README.md [hunk 1]
```diff
@@ -458,38 +469,48 @@ Crates tailored for specific boards.
...
- ![crates.io](https://img.shields.io/crates/v/itsybitsy_m4.svg)
-- [`trinket_m0`](https://crates.io/crates/trinket_m0) - Board support for the [Trinket M0 board](https://www.adafruit.com/product/3500)
- ![crates.io](https://img.shields.io/crates/v/trinket_m0.svg)
-- [`adafruit-feather-rp2040`](https://github.com/rp-rs/rp-hal) - Board Support Crate for the [Adafruit Feather RP2040].
-- [`adafruit-itsy-bitsy-rp2040`](https://github.com/rp-rs/rp-hal) - Board Support Crate for the [Adafruit ItsyBitsy RP2040].
-- [`adafruit-kb2040`](https://github.com/rp-rs/rp-hal) - Board Support Crate for the [Adafruit KB2040].
-- [`adafruit-macropad`](https://github.com/rp-rs/rp-hal) - Board Support Crate for the [Adafruit Macropad].
-- [`adafruit-qt-py-rp2040`](https://github.com/rp-rs/rp-hal) - Board Support Crate for the [Adafruit QT Py RP2040].
-
+- [`metro_m0`](https://crates.io/crates/metro_m0) - Board support for the [Metro M0 board] in the [atsamd-rs] repo. It is a [atsamd-rs tier 1 support] support board. ![crates.io](https://img.shields.io/crates/v/metro_m0.svg)
```
Comment (eldruin) [inline README.md:472]: The word `support` is duplicated. Probably some copy&paste error.

Comment (TDHolmes) [reply README.md:472]: Thanks! fixed up in #358

**Pull request merged**